app/models/user.rb

```diff
@@ -431,6 +431,12 @@ def no_resets?
     has_role?(:no_resets)
   end
 
+  # Whether logged out users can reply to this user's comments with the given
+  # ultimate parent
+  def disallow_guest_replies?(ultimate_parent)
```

Comment:
This works, but I'm not sure if it's the right approach. (Others might want to jump in and tell me I'm totally off base here.)

It's true that the user, not the comment, is the thing that controls reply permissions, so I can definitely see the logic behind putting this check in the User model.

However, we're only going to use this when we already have a comment handy, so it feels overly complicated to always need to do something like `comment.user.disallow_guest_replies?(comment.ultimate_parent)` instead of something like `comment.guest_replies_disallowed?` imo, the way we handled the block helpers (`blocked_by_comment?(comment)` and `blocked_by?(comment)`) sets a precedent for using the comment rather than the user in cases where it might be more convenient.

I think there's also some potential for confusion with this approach -- the code comment helps, but when I first saw the method, it was in the comment helper, and I thought it was checking if the user had the work (ultimate parent) set to disallow guest comments.

Reply:
I think I am in agreement with sarken here.

Reply:
I was debating where to put it, I initially decided User because it's a user preference. Anyway, happy to move it, done.

Now the comment model has two methods with very similar names, one of them used only for validation. I'm not happy with the method names, they don't feel self-explanatory for what the difference is. I could switch back to using validate_each so the validation uses a block instead of a method. Or the validation method could be private, or they could have better names. Suggestions appreciated.